### ISSUE TITLE
Links inside input fields feature request

### ISSUE THREAD

**Issue by Audio951** [mProjectsCode/obsidian-meta-bind-plugin#26]
-   [x] The Plugin is up to date
-   [x] Obsidian is up to date

**Describe the bug**
I'm not sure exactly when or why it became broken, but things were working well for a while.
Currently, as far as I can tell either the bindings are broken in reading mode or it's using stale data.
Sometimes even editing mode seems to not work until I fiddle with it.
Though after messing with it more, I'm beginning to suspect manipulating these at edit time may not be intended?
The input controls sort of work when inside of a table while in edit mode - this may not be intended.

Here's an example of how I'm using the plugin with dataview:

```
|                 |                         |                                                                                                                                                  |
| --------------- | ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| An integer      | `INPUT[text:text1]`     | `$= parseInt(dv.current().file.frontmatter.text1)`                                                                                               |
| Another integer | `INPUT[text:text2]`     | `$= parseInt(dv.current().file.frontmatter.text2)`                                                                                               |
| Add integers?   | `INPUT[toggle:toggle1]` | `$= dv.current().file.frontmatter.toggle1`                                                                                                       |
| Result          |                         | `$= parseInt(dv.current().file.frontmatter.text1) + (dv.current().file.frontmatter.toggle1 ? parseInt(dv.current().file.frontmatter.text2) : 0)` |
```

I'm attempting to create an interactable DnD 5e character sheet.

**To Reproduce**

I haven't been able to replicate the issue on purpose, I can try attaching my vault or something if it would help.

**Occurs on**

-   [x] Windows

**Plugin version**
0.3.2

**Additional Context**
I tried pasting the content of the broken note into a new note and it seems to have the same issue.

**Comment by Audio951** [comment]
After further investigation I believe this may actually be the fault of Dataview. It doesn't seem to update the view when I flip the toggle because the underlying content of the file doesn't change I guess? I thought the metadata changing would trigger a refresh but that does not seem to be the case. So, Dataview is pulling stale cached data effectively, which may be the fault of how obsidian itself works. Will continue looking into it.

**Comment by mProjectsCode** [comment]
Yeah, it might be that dataview is using obsidian's inbuilt metadata cache, which this plugin is also using. But I have found obsidian's metadata cache to be unreliable and I will probably switch to my own parser again

**Comment by mProjectsCode** [comment]
Though without a way to reliably repo the issue it will be hard for me to figure it out. 

I do use dv + meta bind in my own notes and so far I have had no problems.
![image](https://user-images.githubusercontent.com/61602848/202411040-d10fd028-5b3c-4f0e-a7e0-c6404041c85f.png)
![image](https://user-images.githubusercontent.com/61602848/202411244-1d7e8668-4334-4925-84a6-dda1b243ebb0.png)

**Comment by Audio951** [comment]
After even more testing and shenanigans, it seems to be related to having dataview javascript inline properties and inputs in a table inside of a column layout. If I copy the exact same code and just remove the table formatting and columns, it seems to work ok. This is unfortunate because horizontal layouts are such a pain to make. I found a weird workaround where I forced the activeLeaf to rebuild the view after a short delay and technically that kinda works but it's not ideal. I ended up hijacking cleanUpUpdateQueues haha.

**Comment by Audio951** [comment]
First page complete-ish. CSS is a huge mess, but thanks to the power of this plugin I can have editable fields in this reasonably nice looking form.

![image](https://user-images.githubusercontent.com/9119761/203941488-fb16a9ff-d20d-4c2d-b8ae-5a3588bd1a74.png)

**Comment by Audio951** [comment]
Another update and a potential feature request. 

I felt limited by the textarea inputs because they obviously don't handle markdown, so I pulled some shenanigans and added a stripped down version of markdown processing to them:
![image](https://user-images.githubusercontent.com/9119761/204070836-2955e3ad-449d-46ba-96fa-5f5cea55d047.png)

It's like a pseudo-editor. Here's the 'ATTACKS & SPELLCASTING' textarea input in edit mode:
![image](https://user-images.githubusercontent.com/9119761/204070863-e01d596d-f6ff-4d8f-9a89-6181af834933.png)

The benefit being I can hover over links and see the contents for easy referencing which was the original reason I wanted to make a character sheet in obsidian to begin with. See here:
![image](https://user-images.githubusercontent.com/9119761/204070998-45cfc2ac-1208-487d-b606-826a91273554.png)

The potential feature request would be to implement something like this in the plugin itself. 

An issue I'm encountering is while editing the contents of the textarea input it saves and triggers an update that kicks me back out to view mode. Trying to figure out how to work around that but in the mean time I thought you might like to see the idea.

**Comment by mProjectsCode** [comment]
I will remove the bug label for now and add the feature request label, as it seems like the original bug is related to dv and not this plugin

**Comment by Morekon** [comment]
@Audio951 Wow that is awesome. Is it somehow possible that you share some sort of template for that?

**Comment by mayurankv** [comment]
I'd also love to see a template :o

**Comment by Audio951** [comment]
I would not share something like that here without permission - this is not my project. I'm realizing now that despite my excitement, trying to showcase how I've used their plugin to do stuff in a former bug thread might be very distasteful without asking for permission first. Apologies if I did overstep a line in doing so.

**Comment by Audio951** [comment]
Ok, awesome. I created a show and tell post over in discussions for anyone that is interested. 
Thanks!

**Comment by mProjectsCode** [comment]
Feature has been implemented in `0.4.0`
